projects/storefrontlib/cms-components/navigation/scroll-to-top/scroll-to-top.component.ts

```diff
@@ -49,15 +56,32 @@ export class ScrollToTopComponent implements OnInit {
   @Optional() protected featureConfigService = inject(FeatureConfigService, {
     optional: true,
   });
+  @Optional() protected anonymousConsentsService = inject(
+    AnonymousConsentsService,
+    {
+      optional: true,
+    }
+  );
 
   constructor(
     protected winRef: WindowRef,
     protected componentData: CmsComponentData<CmsScrollToTopComponent>,
     protected selectFocusUtility: SelectFocusUtility
-  ) {}
+  ) {
+    useFeatureStyles('a11yScrollToTopPositioning');
+  }
 
   ngOnInit(): void {
     this.setConfig();
+    if (this.featureConfigService?.isEnabled('a11yScrollToTopPositioning')) {
+      this.subscription.add(
+        this.anonymousConsentsService
+          ?.isBannerVisible()
+          .subscribe((visible) => {
```

Comment:
Do you think we could use an observable to set the class value instead of making a subscription? Less code and then we can use the async pipe.

Reply:
Yes, I was able to reimplement it with async pipe utilizing `:has` pseudoselector in scss. Works event better now!